src/webgl/shaders/cubeVertex.vert

```diff
@@ -0,0 +1,13 @@
+// Adapted from:  https://learnopengl.com/PBR/IBL/Diffuse-irradiance
+layout (location = 0) IN vec3 aPosition;
```

Comment:
This looks good, but I feel like do we actually need this layout(). Can it be done just with `IN vec3 aPosition;`

Reply:
Well honestly speaking , I'm also not sure about what exactly it signifies here, but as i searched on web , I found this
If your application code (typically written in a language like C++ or JavaScript) explicitly binds attribute locations using `glBindAttribLocation`, you can omit the `layout(location = 0)` in the shader. The application code would look something like this:
``` js
glBindAttribLocation(program, 0, "aPosition");
```
If your application does not do this, you should keep `layout(location = 0)` to ensure that the attribute location is explicitly defined.

Also , just to mention , I tried running the test sketch by omitting this, and the results were similar , So I think we should keep it for a safer side as its not causing any error.
What say?!

Reply:
Yes, this should not give any errors. Since, vertex shaders are just dealing with the positions, I think we don't have layout(...) at other files and they are working perfectly. 

But the main reason in my mind is, since we only use `layout` with `in` we can't use it with `attributes`, where I think is `layout` used only for webgl-2? If that the case, `layout` should be omitted.